protobuf/lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -64,11 +64,12 @@ class CodedBufferWriter {
     _commitChunk(true);
   }
 
-  void writeField(int fieldNumber, int fieldType, fieldValue) {
+  void writeField(int fieldNumber, int fieldType, Object? fieldValue) {
     final valueType = fieldType & ~0x07;
 
     if ((fieldType & PbFieldType._PACKED_BIT) != 0) {
-      if (!fieldValue.isEmpty) {
+      fieldValue as List<Int64>;
```

Comment:
Can we write this code as 

```
final listValue = fieldValue as List<Object?>;
```

and then use `listValue`? I find such code easier to read. This applies to all other cases.

(also I doubt it's always `List<Int64>`, though I am not sure).

Reply:
I really like the new `fieldValue as List<Object?>;` pattern enabled by flow analysis / type promotion :) it feels strange to convert `fieldValue as List<Object?>;` to `final listValue = fieldValue as List<Object?>;`, because `fieldValue` is still promoted in such an assignment, so no errors pop up. The subsequent uses of `fieldValue` as a `List<Object?>` continue to work, and I must hunt down each reference to `fieldValue`, in the scope of the cast, and change it to refer to `listValue`.

Reply:
FWIW using a variable also allows to satisfy dart2js constraints, you could write 

```
final List<Object?> listValue = fieldValue;
```

to promote at 0 cost for dart2js.